File: domain/src/main/java/com/aceplus/domain/model/customer/Customer.java
package com.aceplus.domain.model.customer;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.io.Serializable;

public class Customer implements Serializable {
    @SerializedName("id")
    @Expose
    private int id;
    @SerializedName("CUSTOMER_ID")
    @Expose
    private String customerId;
    @SerializedName("CUSTOMER_NAME")
    @Expose
    private String customerName;
    @SerializedName("CUSTOMER_TYPE_ID")
    @Expose
    private String customerTypeId;
    @SerializedName("CUSTOMER_TYPE_NAME")
    @Expose
    private String customerTypeName;
    @SerializedName("ADDRESS")
    @Expose
    private String address;
    @SerializedName("PH")
    @Expose
    private String phone;
    @SerializedName("township_number")
    @Expose
    private String township;
    @SerializedName("CREDIT_TERM")
    @Expose
    private String creditTerms;
    @SerializedName("CREDIT_LIMIT")
    @Expose
    private double creditLimit;
    @SerializedName("CREDIT_AMT")
    @Expose
    private double creditAmt;
    @SerializedName("DUE_AMT")
    @Expose
    private double dueAmt;
    @SerializedName("PREPAID_AMT")
    @Expose
    private double prepaidAmt;
    @SerializedName("PAYMENT_TYPE")
    @Expose
    private String paymentType;
    @SerializedName("IS_IN_ROUTE")
    @Expose
    private boolean isInRoute;
    @SerializedName("LATITUDE")
    @Expose
    private double latitude;
    @SerializedName("LONGITUDE")
    @Expose
    private double longitude;
    @SerializedName("VISIT_RECORD")
    @Expose
    private int visitRecord;

    @SerializedName("district_id")
    @Expose
    private int district_id;
    @SerializedName("state_division_id")
    @Expose
    private int state_division_id;
    @SerializedName("contact_person")
    @Expose
    private String contact_person;
    @SerializedName("customer_category_no")
    @Expose
    private String customer_category_no;
    @SerializedName("shop_type_id")
    @Expose
    private int shopTypeId;

    @SerializedName("route_schedule_status")
    @Expose
    private String routeScheduleStatus;

    private int flag;
    private String fax;

    public Customer(){}

    public Customer(String customerId, String customerName, String customerTypeId, String customerTypeName, String address
            , String phone, String township, String creditTerms, double creditLimit, double creditAmt, double dueAmt
            , double prepaidAmt, String paymentType, String isInRoute, double latitude, double longitude, int visitRecord) {

        this.customerId = customerId;
        this.customerName = customerName;
        this.customerTypeId = customerTypeId;
        this.customerTypeName = customerTypeName;
        this.address = address;
        this.phone = phone;
        this.township = township;
        this.creditTerms = creditTerms;
        this.creditLimit = creditLimit;
        this.creditAmt = creditAmt;
        this.dueAmt = dueAmt;
        this.prepaidAmt = prepaidAmt;
        this.paymentType = paymentType;

        if(isInRoute != null && !isInRoute.equals("")) {
            this.isInRoute = isInRoute.equalsIgnoreCase("true");
        }

        this.latitude = latitude;
        this.longitude = longitude;
        this.visitRecord = visitRecord;
    }



    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getCustomerId() {
        return customerId;
    }

    public void setCustomerId(String customerId) {
        this.customerId = customerId;
    }

    public String getCustomerName() {
        return customerName;
    }

    public void setCustomerName(String customerName) {
        this.customerName = customerName;
    }

    public String getCustomerTypeId() {
        return customerTypeId;
    }

    public void setCustomerTypeId(String customerTypeId) {
        this.customerTypeId = customerTypeId;
    }

    public String getCustomerTypeName() {
        return customerTypeName;
    }

    public void setCustomerTypeName(String customerTypeName) {
        this.customerTypeName = customerTypeName;
    }

    public String getAddress() {
        return address;
    }

    public void setAddress(String address) {
        this.address = address;
    }

    public String getPhone() {
        return phone;
    }

    public void setPhone(String phone) {
        this.phone = phone;
    }

    public String getTownship() {
        return township;
    }

    public void setTownship(String township) {
        this.township = township;
    }

    public String getCreditTerms() {
        return creditTerms;
    }

    public void setCreditTerms(String creditTerms) {
        this.creditTerms = creditTerms;
    }

    public double getCreditLimit() {
        return creditLimit;
    }

    public void setCreditLimit(double creditLimit) {
        this.creditLimit = creditLimit;
    }

    public double getCreditAmt() {
        return creditAmt;
    }

    public void setCreditAmt(double creditAmt) {
        this.creditAmt = creditAmt;
    }

    public double getDueAmt() {
        return dueAmt;
    }

    public void setDueAmt(double dueAmt) {
        this.dueAmt = dueAmt;
    }

    public double getPrepaidAmt() {
        return prepaidAmt;
    }

    public void setPrepaidAmt(double prepaidAmt) {
        this.prepaidAmt = prepaidAmt;
    }

    public String getPaymentType() {
        return paymentType;
    }

    public void setPaymentType(String paymentType) {
        this.paymentType = paymentType;
    }

    public boolean isInRoute() {
        return isInRoute;
    }

    public void setInRoute(boolean inRoute) {
        isInRoute = inRoute;
    }

    public double getLatitude() {
        return latitude;
    }

    public void setLatitude(double latitude) {
        this.latitude = latitude;
    }

    public double getLongitude() {
        return longitude;
    }

    public void setLongitude(double longitude) {
        this.longitude = longitude;
    }

    public int getVisitRecord() {
        return visitRecord;
    }

    public void setVisitRecord(int visitRecord) {
        this.visitRecord = visitRecord;
    }

    public int getDistrict_id() {
        return district_id;
    }

    public void setDistrict_id(int district_id) {
        this.district_id = district_id;
    }

    public int getState_division_id() {
        return state_division_id;
    }

    public void setState_division_id(int state_division_id) {
        this.state_division_id = state_division_id;
    }

    public String getContact_person() {
        return contact_person;
    }

    public void setContact_person(String contact_person) {
        this.contact_person = contact_person;
    }

    public String getCustomer_category_no() {
        return customer_category_no;
    }

    public void setCustomer_category_no(String customer_category_no) {
        this.customer_category_no = customer_category_no;
    }

    public int getShopTypeId() {
        return shopTypeId;
    }

    public void setShopTypeId(int shopTypeId) {
        this.shopTypeId = shopTypeId;
    }

    public int getFlag() {
        return flag;
    }

    public void setFlag(int flag) {
        this.flag = flag;
    }

    public String getFax() {
        return fax;
    }

    public void setFax(String fax) {
        this.fax = fax;
    }

    public String getRouteScheduleStatus() {
        return routeScheduleStatus;
    }

    public void setRouteScheduleStatus(String routeScheduleStatus) {
        this.routeScheduleStatus = routeScheduleStatus;
    }
}

File: domain/src/main/java/com/aceplus/domain/entity/customer/Customer.kt
package com.aceplus.domain.entity.customer

import android.arch.persistence.room.ColumnInfo
import android.arch.persistence.room.Entity
import android.arch.persistence.room.PrimaryKey
import android.os.Parcel
import android.os.Parcelable
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName


@Entity(tableName = "customer")
class Customer() : Parcelable {

    @PrimaryKey
    @ColumnInfo(name = "id")
    @SerializedName("id")
    @Expose
    var id: Int = 0

    @ColumnInfo(name = "customer_id")
    @SerializedName("customer_id")
    @Expose
    var customer_id: String? = null

    @ColumnInfo(name = "customer_name")
    @SerializedName("customer_name")
    @Expose
    var customer_name: String? = null

    @ColumnInfo(name = "customer_type_id")
    @SerializedName("customer_type_id")
    @Expose
    var customer_type_id: String? = null

    @ColumnInfo(name = "customer_type_name")
    @SerializedName("customer_type_name")
    @Expose
    var customer_type_name: String? = null

    @ColumnInfo(name = "address")
    @SerializedName("address")
    @Expose
    var address: String? = null

    @ColumnInfo(name = "phone")
    @SerializedName("phone")
    @Expose
    var phone: String? = null

    @ColumnInfo(name = "township")
    @SerializedName("township")
    @Expose
    var township: String? = null

    @ColumnInfo(name = "credit_term")
    @SerializedName("credit_term")
    @Expose
    var credit_term: String? = null

    @ColumnInfo(name = "credit_limit")
    @SerializedName("credit_limit")
    @Expose
    var credit_limit: String? = null

    @ColumnInfo(name = "credit_amount")
    @SerializedName("credit_amount")
    @Expose
    var credit_amount: String? = null

    @ColumnInfo(name = "due_amount")
    @SerializedName("due_amount")
    @Expose
    var due_amount: String? = null

    @ColumnInfo(name = "prepaid_amount")
    @SerializedName("prepaid_amount")
    @Expose
    var prepaid_amount: String? = null

    @ColumnInfo(name = "payment_type")
    @SerializedName("payment_type")
    @Expose
    var payment_type: String? = null

    @ColumnInfo(name = "is_in_route")
    @SerializedName("is_in_route")
    @Expose
    var in_route: String? = null

    @ColumnInfo(name = "latitude")
    @SerializedName("latitude")
    @Expose
    var latitude: String? = null

    @ColumnInfo(name = "longitude")
    @SerializedName("longitude")
    @Expose
    var longitude: String? = null

    @ColumnInfo(name = "visit_record")
    @SerializedName("visit_record")
    @Expose
    var visit_record: String? = null

    @ColumnInfo(name = "district_id")
    @SerializedName("district_id")
    @Expose
    var district_id: Int = 0

    @ColumnInfo(name = "state_division_id")
    @SerializedName("state_division_id")
    @Expose
    var state_division_id: Int = 0

    @ColumnInfo(name = "shop_type_id")
    @SerializedName("shop_type_id")
    @Expose
    var shop_type_id: Int = 0

    @ColumnInfo(name = "street_id")
    @SerializedName("street_id")
    @Expose
    var street_id: Int = 0

    @ColumnInfo(name = "fax")
    @SerializedName("fax")
    @Expose
    var fax: String? = null

    @ColumnInfo(name = "township_number")
    @SerializedName("township_number")
    @Expose
    var township_number: String? = null

    @ColumnInfo(name = "customer_category_no")
    @SerializedName("customer_category_no")
    @Expose
    var customer_category_no: String? = null

    @ColumnInfo(name = "contact_person")
    @SerializedName("contact_person")
    @Expose
    var contact_person: String? = null

    @ColumnInfo(name = "route_schedule_status")
    @SerializedName("route_schedule_status")
    @Expose
    var route_schedule_status: String? = null

    @ColumnInfo(name = "created_user_id")
    @SerializedName("created_user_id")
    @Expose
    var created_user_id: String? = null

    @ColumnInfo(name = "created_date")
    @SerializedName("created_date")
    @Expose
    var created_date: String? = null

    @ColumnInfo(name = "flag")
    @SerializedName("flag")
    @Expose
    var flag: String? = null

    constructor(parcel: Parcel) : this() {
        id = parcel.readInt()
        customer_id = parcel.readString()
        customer_name = parcel.readString()
        customer_type_id = parcel.readString()
        customer_type_name = parcel.readString()
        address = parcel.readString()
        phone = parcel.readString()
        township = parcel.readString()
        credit_term = parcel.readString()
        credit_limit = parcel.readString()
        credit_amount = parcel.readString()
        due_amount = parcel.readString()
        prepaid_amount = parcel.readString()
        payment_type = parcel.readString()
        in_route = parcel.readString()
        latitude = parcel.readString()
        longitude = parcel.readString()
        visit_record = parcel.readString()
        district_id = parcel.readInt()
        state_division_id = parcel.readInt()
        shop_type_id = parcel.readInt()
        street_id = parcel.readInt()
        fax = parcel.readString()
        township_number = parcel.readString()
        customer_category_no = parcel.readString()
        contact_person = parcel.readString()
        route_schedule_status = parcel.readString()
        created_user_id = parcel.readString()
        created_date = parcel.readString()
        flag = parcel.readString()
    }

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeInt(id)
        parcel.writeString(customer_id)
        parcel.writeString(customer_name)
        parcel.writeString(customer_type_id)
        parcel.writeString(customer_type_name)
        parcel.writeString(address)
        parcel.writeString(phone)
        parcel.writeString(township)
        parcel.writeString(credit_term)
        parcel.writeString(credit_limit)
        parcel.writeString(credit_amount)
        parcel.writeString(due_amount)
        parcel.writeString(prepaid_amount)
        parcel.writeString(payment_type)
        parcel.writeString(in_route)
        parcel.writeString(latitude)
        parcel.writeString(longitude)
        parcel.writeString(visit_record)
        parcel.writeInt(district_id)
        parcel.writeInt(state_division_id)
        parcel.writeInt(shop_type_id)
        parcel.writeInt(street_id)
        parcel.writeString(fax)
        parcel.writeString(township_number)
        parcel.writeString(customer_category_no)
        parcel.writeString(contact_person)
        parcel.writeString(route_schedule_status)
        parcel.writeString(created_user_id)
        parcel.writeString(created_date)
        parcel.writeString(flag)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<Customer> {
        override fun createFromParcel(parcel: Parcel): Customer {
            return Customer(parcel)
        }

        override fun newArray(size: Int): Array<Customer?> {
            return arrayOfNulls(size)
        }
    }
}